pkg/client/build.go

```diff
@@ -491,6 +516,11 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 	}
 	defer c.docker.ImageRemove(context.Background(), ephemeralBuilder.Name(), types.RemoveOptions{Force: true})
 
+	builderOS, err := rawBuilderImage.OS()
+	if err != nil {
+		return fmt.Errorf("failed to get builder OS: %w", err)
+	}
```

Comment:
I think we could also do `targetToUse.OS`

Reply:
Interesting, you moved the validation way too far, is it for any particular reason?

Reply:
It's because we are calling `getTargetFromBuilder` where we were calling `rawBuilderImage.OS()` before